cli/internal/state/state.go

```diff
@@ -7,23 +7,126 @@ SPDX-License-Identifier: AGPL-3.0-only
 // package state defines the structure of the Constellation state file.
 package state
 
+import (
+	"fmt"
+
+	"dario.cat/mergo"
+	"github.com/edgelesssys/constellation/v2/cli/internal/clusterid"
+	"github.com/edgelesssys/constellation/v2/internal/config"
+	"github.com/edgelesssys/constellation/v2/internal/file"
+)
+
 const (
 	// Version1 is the first version of the state file.
 	Version1 = "v1"
 )
 
+// ReadFromFile reads the state file at the given path and returns the state.
+func ReadFromFile(fileHandler file.Handler, path string) (*State, error) {
+	state := &State{}
+	if err := fileHandler.ReadYAML(path, &state); err != nil {
+		return nil, fmt.Errorf("reading state file: %w", err)
+	}
+	return state, nil
+}
+
 // State describe the entire state to describe a Constellation cluster.
 type State struct {
 	Version        string         `yaml:"version"`
 	Infrastructure Infrastructure `yaml:"infrastructure"`
+	ClusterValues  ClusterValues  `yaml:"clusterValues"`
 }
 
-// NewState creates a new state with the given infrastructure.
-func NewState(Infrastructure Infrastructure) State {
-	return State{
-		Version:        Version1,
-		Infrastructure: Infrastructure,
+// New creates a new cluster state (file).
+func New() *State {
+	return &State{
+		Version: Version1,
+	}
+}
+
+// NewFromIDFile creates a new cluster state file from the given ID file.
+func NewFromIDFile(idFile clusterid.File) *State {
+	s := New()
+
+	if idFile.ClusterID != "" {
+		s.ClusterValues.ClusterID = idFile.ClusterID
+	}
```

Comment:
Since this function creates a new `State`, and `idFile` is not a pointer, you can make most of the assignments here unconditionally.